dashboard/app/controllers/admin_users_controller.rb

```diff
@@ -81,7 +81,12 @@ def manual_pass
     unless user
       flash[:alert] = "User (ID: #{params[:user_id]}) not found"
     end
-    script = Script.get_from_cache(params[:script_id_or_name])
+    script = nil
+    if params[:script_id_or_name].to_i.to_s == params[:script_id_or_name]
+      script = Script.find_by_id(params[:script_id_or_name])
+    else
+      script = Script.find_by_name(params[:script_id_or_name])
+    end
```

Comment:
Yes. It means that Josh S will need to wait a half second longer. And the O(10) times a week it gets used, we fire off one dB query rather than hitting the cache.